From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sun, 21 Nov 2021 16:03:52 +0100
Subject: [PATCH] {BUG in 5.16-rc1} arch/x86/tools/relocs: use
 IS_BUILTIN(CONFIG_FW_LOADER)

clang on x86_64 reports that CONFIG_FW_LOADER_BUILTIN does not exist:

    arch/x86/tools/relocs.c:71:5: error: 'CONFIG_FW_LOADER_BUILTIN' is
    not defined, evaluates to 0 [-Werror,-Wundef]
    #if CONFIG_FW_LOADER_BUILTIN
        ^
---
 arch/x86/tools/relocs.c | 2 +-
 arch/x86/tools/relocs.h | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/x86/tools/relocs.c b/arch/x86/tools/relocs.c
index c736cf2ac76b..e810b5a1dfce 100644
--- a/arch/x86/tools/relocs.c
+++ b/arch/x86/tools/relocs.c
@@ -68,7 +68,7 @@ static const char * const sym_regex_kernel[S_NSYMTYPES] = {
 	"(__parainstructions|__alt_instructions)(_end)?|"
 	"(__iommu_table|__apicdrivers|__smp_locks)(_end)?|"
 	"__(start|end)_pci_.*|"
-#if CONFIG_FW_LOADER_BUILTIN
+#if IS_BUILTIN(CONFIG_FW_LOADER)
 	"__(start|end)_builtin_fw|"
 #endif
 	"__(start|stop)___ksymtab(_gpl)?|"
diff --git a/arch/x86/tools/relocs.h b/arch/x86/tools/relocs.h
index e71321f1f5a9..ed0929418710 100644
--- a/arch/x86/tools/relocs.h
+++ b/arch/x86/tools/relocs.h
@@ -16,6 +16,7 @@
 #include <endian.h>
 #include <regex.h>
 #include <tools/le_byteshift.h>
+#include <tools/config.h>
 
 __attribute__((__format__(printf, 1, 2)))
 void die(const char *fmt, ...) __attribute__((noreturn));
-- 
